apps/condo/domains/miniapp/access/B2BAppPermission.js

```diff
@@ -2,28 +2,43 @@
  * Generated by `createschema miniapp.B2BAppPermission 'app:Relationship:B2BApp:PROTECT; key:Text'`
  */
 
+const compact = require('lodash/compact')
 const get = require('lodash/get')
+const uniq = require('lodash/uniq')
 
 const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
-const { getById } = require('@open-condo/keystone/schema')
+const { getById, find } = require('@open-condo/keystone/schema')
 
 const { checkB2BAppAccessRight } = require('@condo/domains/miniapp/utils/accessSchema')
+const { queryOrganizationEmployeeFor } = require('@condo/domains/organization/utils/accessSchema')
 const { SERVICE } = require('@condo/domains/user/constants/common')
 
+
 /**
  * B2B App permissions may only be read:
  * 1. Admin / support
  * 2. Integration service account
  * Employee cannot access permissions, since they interact only with their B2B app roles
  */
-async function canReadB2BAppPermissions ({ authentication: { item: user } }) {
+async function canReadB2BAppPermissions ({ authentication: { item: user } }, context) {
     if (!user) return throwAuthenticationError()
     if (user.deletedAt) return false
 
     if (user.isAdmin || user.isSupport) return {}
 
+    const userOrganizations = await find('Organization', queryOrganizationEmployeeFor(user.id))
+    const b2bAppContexts =  await find('B2BAppContext', {
+        organization: { id_in: userOrganizations.map(organization => organization.id) },
+        deletedAt: null,
+    })
+    const b2bAppIds = uniq(compact(b2bAppContexts.map(context => get(context, 'app'))))
 
-    return { app: { accessRights_some: { user: { id: user.id }, deletedAt: null } } }
+    return {
+        OR: [
+            { app: { accessRights_some: { user: { id: user.id }, deletedAt: null } } },
```

Comment:
Split this `or` condition

User with type service have one query, staff another, residents is banned